### PR TITLE
Move non-app projects out of 'apps' and into 'libraries'

### PR DESCRIPTION
**Pull request by iclanton**
This PR specifically moves `@microsoft/rush-lib` and `@microsoft/api-extractor-model` to the `libraries` folder.

### CODE REVIEW

**Comment by octogonz** [comment]
https://github.com/microsoft/rushstack/pull/3321 is about to get merged tomorrow. Could we do this afterwards to avoid creating a merge conflict?

**Comment by iclanton** [comment]
> #3321 is about to get merged tomorrow. Could we do this afterwards to avoid creating a merge conflict?

Sure

File: common/config/azure-pipelines/templates/build.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ steps:
       # Prevent time-based browserslist update warning
       # See https://github.com/microsoft/rushstack/issues/2981
       BROWSERSLIST_IGNORE_OLD_DATA: 1
-  - script: 'node apps/rush-lib/lib/start.js build --verbose --production'
+  - script: 'node libraries/rush-lib/lib/start.js build --verbose --production'
```
Comment (dmichon-msft) [inline common/config/azure-pipelines/templates/build.yaml:21]:
```suggestion
  - script: 'node apps/rush/lib/start-dev.js build --verbose --production --timeline'
```

**Review by dmichon-msft — APPROVED** [review]

**Review by octogonz — APPROVED** [review]
PR #3321 has been merged, so you're good to proceed with this PR.  (Double-check that there isn't a merge conflict with those changes.)

File: libraries/api-extractor-model/package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/microsoft/rushstack.git",
-    "directory": "apps/api-extractor-model"
+    "directory": "libraries/api-extractor-model"
```
Comment (octogonz) [inline libraries/api-extractor-model/package.json:8]: We'll need a corresponding fix in https://github.com/microsoft/api-extractor.com-website/

File: libraries/rush-lib/package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/microsoft/rushstack.git",
-    "directory": "apps/rush-lib"
+    "directory": "libraries/rush-lib"
```
Comment (octogonz) [inline libraries/rush-lib/package.json:8]: We'll need a corresponding fix in https://github.com/microsoft/rushstack-websites

Comment (iclanton) [reply libraries/rush-lib/package.json:8]: https://github.com/microsoft/rushstack-websites/pull/45

File: libraries/api-extractor-model/package.json (hunk 2 above)
Comment (iclanton) [reply libraries/api-extractor-model/package.json:8]: https://github.com/microsoft/api-extractor.com-website/pull/42/

**Comment by zelliott** [comment]
Does this also need to be renamed from `apps/` to `libraries/`?

https://github.com/microsoft/rushstack/blob/c7aef037ee57dbb0892b2882c9d8404cc51ce0fc/.vscode/launch.json#L28

**Pull request merged**